snappiershot/snapshot/snapshot.py

```diff
@@ -90,6 +91,54 @@ def assert_match(self, value: Any, exact: bool = False, update: bool = False) ->
         self._snapshot_file.mark_passed(current_index)
         return True
 
+    def raises(
+        self, expected_exception: _ExceptionTypes, *, update: bool = False
+    ) -> "_RaisesContext":
+        """ Assert that a code block raises an expected exception
+        and snapshot the value of the raised exception.
+
+        This is directly inspired by the ``pytest.raises`` method.
+
+        This method must be used as a context manager:
+        ```python
+        >>> snapshot: Snapshot
+        >>> with snapshot.raises(ValueError):
+        ...    raise ValueError("This message will be part of the snapshot. ")
+        ```
+
+        Args:
+            expected_exception: BaseException type or tuple of BaseException types
+              used to assert against the type of the raised exception.
+            update: Whether to update the snapshot associated with the raised exception.
+
+        Raises:
+            TypeError: If the arguments to this method have invalid types.
+        """
+        # Type checking and type coercion of arguments.
```

Comment:
this type checking and input coersion should perhaps live in its own private helper method